internal/telemetry/client.go

```diff
@@ -153,9 +153,7 @@ func log(msg string, args ...interface{}) {
 // DD_TELEMETRY_HEARTBEAT_INTERVAL, DD_INSTRUMENTATION_TELEMETRY_DEBUG,
 // and DD_TELEMETRY_DEPENDENCY_COLLECTION_ENABLED.
 // TODO: implement passing in error information about tracer start
-func (c *client) Start(configuration []Configuration) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
```

Comment:
Why was this locked before but doesn't need to be now?

Reply:
Now that `ProductStart/Stop` is exported and used by the tracer and profiler, the lock will be held there. I'll add a comment that `start` should be called with `c.mu` locked to clarify why this was removed